common/include/pcl/pcl_macros.h

```diff
@@ -118,11 +118,11 @@ namespace pcl
   using int_fast16_t PCL_DEPRECATED("use std::int_fast16_t instead of pcl::int_fast16_t") = std::int_fast16_t;
 }
 
-#if defined _WIN32 && defined _MSC_VER
+#if defined _WIN32
 
 // Define math constants, without including math.h, to prevent polluting global namespace with old math methods
 // Copied from math.h
-#ifndef _MATH_DEFINES_DEFINED
+#if !defined _MATH_DEFINES_DEFINED && !defined M_2_SQRTPI
```

Comment:
Any reason for separate check for `M_2_SQRTPI`?

Reply:
Because it seems like mingw's cmath header doesn't define _MATH_DEFINES_DEFINED. At least I can't get it to work reliably. There might be a better way to solve this though.

Reply:
In that case, please add a comment 😄